core-bundle/src/Routing/Content/StringResolver.php

```diff
@@ -36,6 +38,12 @@ public function resolve(object $content): ContentUrlResult|null
             $url = $this->urlHelper->getAbsoluteUrl($url);
         }
 
+        // Resolve protocol-relative URLs that are ignored by the UrlHelper
+        if (str_starts_with($url, '//')) {
+            $protocol = $this->requestStack->getCurrentRequest()?->getScheme() ?? 'http';
```

Comment:
```suggestion
            $protocol = $this->requestStack->getCurrentRequest()?->getScheme() ?? 'https';
```
Shouldn’t the default be HTTPS?

Reply:
I don't think we can assume that?

Reply:
I think we can.

Reply:
As HTTPS is also the default in most browsers nowadays, I think we should default to HTTPS everywhere we can.

Reply:
Browsers seem to use `https` as default (see https://blog.chromium.org/2021/03/a-safer-default-for-navigation-https.html), but they fall back to `http`. They would not fall back in our case, right?

Reply:
> They would not fall back in our case, right?

For explicit URLs (not a domain entered by the user) they do not fallback one way or the other (with some exceptions like HSTS).

In practice this fallback will never be used anyways. I just thought that we should generally fallback to `https` for security reasons.

Maybe the following would be a better choice here?

```suggestion
            $protocol = $this->requestStack->getCurrentRequest()?->getScheme() ?? throw new RuntimeException('Unable to generate absolute URL because of missing request scheme.');
```

Reply:
I don't understand. ANY website supports http, MOST websites support https and redirect to https if you access it on http. Why would it be better to error or use a protocol that is possibly not supported, instead of using the one we KNOW is supported. It _might_ cause a redirect, but **it works!**?? 🤷

Reply:
> Why would it be better to error…

Because a URL starting with `//` cannot be resolved if there is no current request. If this happens, there is a bug somewhere that needs to be fixed.

> …or use a protocol that is possibly not supported…

To avoid using a protocol that is guaranteed to be dangerous. Using `http` as a fallback here would be a prefect opportunity for an attacker as they just need to somehow manage to inject `//`

> It _might_ cause a redirect, but **it works!**?? 🤷

Works for the attacke**r!**?? 🤷

Reply:
well I disagree 🤷

Reply:
Let‘s just throw the exception in this case, then we don‘t have to guess a protocol.